### ISSUE TITLE
Whitelist for bot connector service

### ISSUE THREAD

**Issue by alphabj** [Microsoft/BotBuilder#1318]
In order to make my bot as secure as possible, I'd like to disallow any inbound traffic unless it's coming from the MS bot connector. Is there a whitelist of the IPs/domains used? I couldn't find an answer to this question in support and wasn't sure how to reach out to MS. Thanks.

**Comment by brandonh-msft** [comment]
If you [wire up Application Insights to the bot site](http://aihelpwebsite.com/Blog/EntryId/11/Using-Application-Insights-With-Microsoft-Bot-Framework), it will log the traffic from "Dependencies" - in this case that ends up being the bot framework.
Apologies, I don't have any traffic handy right now or I'd give you an example. At any rate that might be a good starting point for what to let in/out.

**Comment by dandriscoll** [comment]
Hi @alphabj, we do not publish a list, and I would encourage you to be resilient about source IPs as we add new servers and services all the time.

The most robust way to ensure a Bot Framework service is authentic is to perform the verification steps described in our authentication guide (https://docs.botframework.com/en-us/restapi/authentication/) or use the Bot Builder SDK, which implements all of those checks.

**Comment by alphabj** [comment]
Thanks guys. I am using the Bot Builder SDK, but would still like to lock it down as much as I can for now. I followed Brandon's lead and came up with these, at least in the case of skype:

https://login.microsoftonline.com
https://api.aps.skype.com
https://skype.botframework.com

I expect I'll need to check again once I set up Twillio.

**Comment by tomlm** [comment]
yeah, again as dan pointed out, every time we deploy new servers in new data centers your bot will break.  SSL + our app signing is much more secure and resilient way of securing Service to service validation.

**Comment by dividor** [comment]
Thanks for all the comments, but just to add ... we're using the SDK, but we cannot open a new port unless we whitelist. I think this is pretty standard in a lot of corporations.

**Comment by nwhitmont** [comment]
@dividor You are commenting on a closed issue. Please open a new issue if you have a question or problem. Thx! 💯
